### DISCUSSION TITLE
Statement sectioning issue for ISNULL

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#109] [General]
Hi Bruce,
Me again.

I was trying this simple query
`select isnull(a, 0) * isnull(b, 0) / 100 as Result from myTable`

And I found this statement has been separated into two statements. at the second ")" instead of one. Please see my output.
```
Start...
Statement 1:  select isnull ( a , 0 ) * isnull ( b , 0
        Token type: Keyword     Text: select
        Token type: SystemIdentifier    Text: isnull
        Token type: Character   Text: (
        Token type: Identifier  Text: a
        Token type: Character   Text: ,
        Token type: NumericLiteral      Text: 0
        Token type: Character   Text: )
        Token type: Operator    Text: *
        Token type: SystemIdentifier    Text: isnull
        Token type: Character   Text: (
        Token type: Identifier  Text: b
        Token type: Character   Text: ,
        Token type: NumericLiteral      Text: 0

Statement 2:  ) as Result from mytable
        Token type: Character   Text: )
        Token type: Keyword     Text: as
        Token type: Identifier  Text: Result
        Token type: Keyword     Text: from
        Token type: Identifier  Text: mytable
```
This may well be a bug and does cause some issues when executing the separated statements.

Thank you for looking into all the issues. I am feeling a bit guilty now.

Regards
Paul

**Comment by bruce-dunwiddie** [comment]
Thank you.

Keep them coming. I've so far been surprised by some of the bigger holes I still had with this new expression parsing.

In theory, the more bugs that get resolved now in SELECT's should make the eventual next phase with all the JOIN and WHERE parsing easy.

**Reply by cptb** [reply]
Your responses are getting quicker and quicker.
I certainly have loads of complex queries, some of them are over 1000 lines. So in the future, more weird scenarios may surface too.

**Reply by bruce-dunwiddie** [reply]
I'm rarely far from a keyboard.

**Comment by bruce-dunwiddie** [comment]
I've verified this issue.

It has to do with the handling of the multi purpose character * in TSQL in the current parsing implementation. The issue does not exist if you change it to a - for example.

The funny thing is that the fix appears to be to delete a line in the parsing logic, for a special case with *, that with all my recent fixes no longer appears to be needed. It's always good when you can delete a branch of parsing logic.

So this is an easy fix. I'll roll it out with the fix for the other issue in a couple of days.

**Reply by bruce-dunwiddie** [reply]
Please verify that it's fixed in https://www.nuget.org/packages/TSQL.Parser/2.5.0.

**Reply by cptb** [reply]
Thank you very much. We just installed the latest version and can say it's working great.
